Serializer/JMSHandlerRegistry.php

```diff
@@ -51,11 +53,18 @@ public function registerHandler($direction, $typeName, $format, $handler): void
      */
     public function getHandler($direction, $typeName, $format)
     {
+        $first = true;
         do {
             $handler = $this->registry->getHandler($direction, $typeName, $format);
             if (null !== $handler) {
+                if (!$first) {
+                    @trigger_error(sprintf('Relying on the custom registry %s to inherit the JMS handler of type `%s` is deprecated since FOSRestBundle 3.1. It will be removed in version 4.0. Use the option `fos_rest.serializer.disable_custom_jms_registry` to disable it.', __CLASS__, $typeName), E_USER_DEPRECATED);
```

Comment:
Are we sure this deprecation will always be triggered? I could imagine that's not the case if `$first` never becomes `false`. I tend to think that triggering a deprecation when the option is set to `false` and here on top of the class for everyone that uses it directly would be the better approach.

Reply:
If the deprecation is not triggered it is either because
- The handler returned is the one expected (the one returned by the jms registry)
- No handler is returned and the behavior is thus the same as the jms registry

I wanted to deprecate is relying on the behavior of this class, so as long as the result is the same as what would jms return I believe it's ok to not trigger a deprecation.
But if we want to warn users they could improve performance by disabling this class, triggering a deprecation when `$first` would indeed be legit.

So in my opinion it depends whether we want to warn users about the performance improvement possibility or not. I think it is better to privilege DX but I would appreciate having your opinion on this point :)

Reply:
I think we should always trigger the deprecation to make users aware of the performance penalty this may have.

Reply:
ok, it's updated :)